ddtrace/internal/telemetry/writer.py

```diff
@@ -552,6 +536,20 @@ def periodic(self):
         if logs_metrics:
             self._generate_logs_event(logs_metrics)
 
+        if self._is_periodic and force_flush is False:
```

Comment:
the above should happen/be flushed every 10 seconds, but the below only once every 60 seconds?

a comment here is going to be really important. I can easily see someone putting flushing of new data in the wrong location and something not working correctly.

also, what happens if the heartbeat interval is not divisible by 10, is that ok?

maybe the stuff that happens less frequently should be moved to it's own function fro clarity?

Reply:
> the above should happen/be flushed every 10 seconds, but the below only once every 60 seconds?
Yup. We aligned on this spec across all tracer teams a few weeks ago 


> what happens if the heartbeat interval is not divisible by 10, is that ok?

We will round up. Ex: If the heartbeat is set to 12 seconds, telemetry events will be sent every 20. I think this is fine. The heartbeat interval does not have to exact. The key thing here is that metrics are aggregated into buckets with a max size of 10 seconds.

To be honest I don't see a valid case for including `DD_TELEMETRY_HEARTBEAT` in our public api. This is not a user facing feature and this configuration is only used in system-tests. We should honestly make it private but this would involve updating system tests and changing this environment variable across all tracers. It may not be worth the effort. 

`I will add better documentation`